components/Button.tsx

```diff
@@ -167,41 +194,56 @@ const StyledButton = styled(DEFAULT_TAG, {
     state: {
       active: {
         backgroundColor: '$gray300',
-        boxShadow: 'inset 0 0 0 1px $gray700',
+        boxShadow: 'inset 0 0 0 1px $colors$gray700',
         color: '$gray900',
-        '&:hover': {
-          backgroundColor: '$gray400',
-          boxShadow: 'inset 0 0 0 1px $gray700',
+        "@media (any-hover: hover)": {
+          '&:hover': {
+            backgroundColor: '$gray400',
+            boxShadow: 'inset 0 0 0 1px $colors$gray700',
+          }
         },
         '&:active': {
           backgroundColor: '$gray400',
         },
         '&:focus': {
-          boxShadow: 'inset 0 0 0 1px $gray700, 0 0 0 1px $gray700',
+          boxShadow: 'inset 0 0 0 1px $colors$gray700, 0 0 0 1px $colors$gray700',
         },
       },
       waiting: {
         backgroundColor: '$gray300',
-        boxShadow: 'inset 0 0 0 1px $gray700',
+        boxShadow: 'inset 0 0 0 1px $colors$gray700',
         color: 'transparent',
         pointerEvents: 'none',
-        '&:hover': {
-          backgroundColor: '$gray400',
-          boxShadow: 'inset 0 0 0 1px $gray700',
+        "@media (any-hover: hover)": {
+          '&:hover': {
+            backgroundColor: '$gray400',
+            boxShadow: 'inset 0 0 0 1px $colors$gray700',
+          }
         },
         '&:active': {
           backgroundColor: '$gray400',
         },
         '&:focus': {
-          boxShadow: 'inset 0 0 0 1px $gray700',
+          boxShadow: 'inset 0 0 0 1px $colors$gray700',
         },
       },
     },
   },
+  defaultVariants: {
+    size: '1',
+    variant: 'gray',
+  },
 });
 
+type ButtonCSSProp = { css?: CSS };
+// TODO: Remove omit fix when this is merged https://github.com/modulz/stitches/issues/421
+type ButtonVariants = Omit<StitchesVariants<typeof StyledButton>, 'size'>;
+type ButtonOwnProps = ButtonCSSProp & ButtonVariants & { size?: any };
+
 type ButtonComponent = Polymorphic.ForwardRefComponent<typeof DEFAULT_TAG, ButtonOwnProps>;
 
 export const Button = React.forwardRef((props, forwardedRef) => {
   return <StyledButton {...props} ref={forwardedRef} />;
 }) as ButtonComponent;
+
+Button.toString = () => `.${StyledButton.className}`;
```

Comment:
Why/when is this needed?

Reply:
This is needed if you need to reference a "non-stitches" component in the Style Object. The `ControlGroup` does